### PR TITLE
Update gradle config

### PR DESCRIPTION
**Pull request by CPUcontrol**

### CODE REVIEW

File: circleci/templates/gradle/config_template-v0_1.yml [hunk 1]
```diff
@@ -1,159 +1,158 @@
...
-            echo "export ORG_GRADLE_PROJECT_dependencyCheckUrlBase='http://10.101.190.7:10179/nvdcve-1.0-%d.json.gz'" >> $BASH_ENV
             source $BASH_ENV
       - checkout
       - run:
-          name: Building
-          command: |
-            ./gradlew build --info
-      - run:
-          name: Save test results
+          name: Maven install without tests
```
Comment (j-fontaine) [inline circleci/templates/gradle/config_template-v0_1.yml]: The name field here still references Maven

File: circleci/templates/gradle/config_template-v0_1.yml [hunk 2]
```diff
@@ -1,159 +1,158 @@
...
           command: |
-            mkdir -p ~/test-results/junit/
-            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} ~/test-results/junit/ \;
-          when: always
-      - store_test_results:
-          path: ~/test-results
-      - store_artifacts:
-          path: ~/test-results/junit
-  build:
+            ./gradlew install -B -DskipTests $DISABLE_DOWNLOAD_PROGRESS_OPTS
```
Comment (j-fontaine) [inline circleci/templates/gradle/config_template-v0_1.yml]: The $DISABLE_DOWNLOAD_PROGRESS_OPTS environment variable generates maven plugin logger guidance. I suspect it won't work properly for gradle.  Have you run this manually to confirm it works as expected?

The same concern exists for me around the -DskipTests command.  We should review and have some guidance around the gradle file and the different stages it allows.

File: circleci/templates/gradle/config_template-v0_1.yml [hunk 3]
```diff
@@ -1,159 +1,158 @@
...
     steps:
       - run: *env
       - checkout
       - run:
-          name: Building
-          command: |
-            ./gradlew clean build
-      - run:
-          name: Save test results
+          name: Maven test phase
```
Comment (j-fontaine) [inline circleci/templates/gradle/config_template-v0_1.yml]: Ref: Maven (see above)

File: circleci/templates/gradle/config_template-v0_1.yml [hunk 4]
```diff
@@ -1,159 +1,158 @@
...
-            mkdir -p ~/test-results/junit/
-            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} ~/test-results/junit/ \;
-          when: always
-      - store_test_results:
-          path: ~/test-results
-      - store_artifacts:
-          path: ~/test-results/junit
-  #Runs OWASP on build
-  owasp:
+            ./gradlew test -B $DISABLE_DOWNLOAD_PROGRESS_OPTS
```
Comment (j-fontaine) [inline circleci/templates/gradle/config_template-v0_1.yml]: REF: $DISABLED_DOWNLOAD_PROGRESS_OPTS (see above)

File: circleci/templates/gradle/config_template-v0_1.yml [hunk 5]
```diff
@@ -1,159 +1,158 @@
...
-            then
-              git checkout $RELEASE_COMMIT
-              git remote add ssh-origin git@github.com:connexta/$CIRCLE_PROJECT_REPONAME.git
-              echo ssh -i $GITHUB_KEY -l git -o StrictHostKeyChecking=no \\"\\$@\\" > run_ssh.sh
-              chmod +x run_ssh.sh
-              git push ssh-origin HEAD:$CIRCLE_BRANCH && git push ssh-origin $RELEASE_TAG
-            fi
-  deploy:
-    docker: *image
+            echo $SONAR_PARAMS;
```
Comment (j-fontaine) [inline circleci/templates/gradle/config_template-v0_1.yml:87]: While this structure works for maven, I'm not certain it works for gradle.  Have you tested this?
We may need to add additional logic to allow sonar to work properly with Gradle that isn't a direct port over.

**Review by j-fontaine — CHANGES_REQUESTED** [review]

File: circleci/templates/gradle/config_template-v0_1.yml (hunk 2 above)
Comment (aaronilovici) [inline circleci/templates/gradle/config_template-v0_1.yml]: -B (batch mode) also doesn't work. You should try these commands before putting up the PR.

File: circleci/templates/gradle/config_template-v0_1.yml [hunk 6]
```diff
@@ -16,18 +16,18 @@ jobs: #Defines the overall Jobs for this particular pipeline to be used in workf
             source $BASH_ENV
       - checkout
       - run:
-          name: Maven install without tests
+          name: Gradle install without tests
           command: |
-            ./gradlew install -B -DskipTests $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            ./gradlew clean build --info
```
Comment (j-fontaine) [inline circleci/templates/gradle/config_template-v0_1.yml]: Does this skip or execute the test phase?
The general idea with maven is that we skip the unit tests and defer them out to the next Unit Testing phase.

File: circleci/templates/gradle/config_template-v0_1.yml (hunk 5 above)
Comment (j-fontaine) [reply circleci/templates/gradle/config_template-v0_1.yml:87]: Do we know if the command line arguments also still work with this or do they get ignored?

**Review by j-fontaine — APPROVED** [review]

**Pull request merged**